atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -261,16 +277,14 @@ def forward_composes(self):
         repos_by_arch = {}
         # set overrides by arch if arches are available
         for compose_info in self.composes_info:
-            for arch in compose_info.get('arches') or []:
-                repos_by_arch.setdefault(arch, [])
-                repos_by_arch[arch].append(compose_info['result_repofile'])
-
+            if compose_info['source_type'] == 'pulp':
+                arch = compose_info['arches'][0]
```

Comment:
Instead of picking the first one, can we trust the value of `arches` in each `compose_info`?

```python
from collections import defaultdict


repos_by_arch = defaultdict(list)

for compose_info in self.composes_info:
  for arch in compose_info['arches']:
    repos_by_arch[arch].append(compose_info['result_repofile'])

for arch, repofiles in repos_by_arch.items():
  override_build_kwarg(self.workflow, 'yum_repourls', repofiles, arch)
```

Reply:
Fixed.